test/unit/org/apache/cassandra/tools/nodetool/DataPathsTest.java

```diff
@@ -0,0 +1,169 @@
+        assertThat(StringUtils.countMatches(tool.getStdout(), "Keyspace :")).isGreaterThan(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tTable :")).isGreaterThan(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tPaths :")).isGreaterThan(1);
+    }
+
+    @Test
+    public void testSelectedKeyspace()
+    {
+        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool(SUBCOMMAND, "system_traces");
+        tool.assertOnCleanExit();
+        assertThat(tool.getStdout()).contains("Keyspace : system_traces");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "Keyspace :")).isEqualTo(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tTable :")).isGreaterThan(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tPaths :")).isGreaterThan(1);
+    }
+
+    @Test
+    public void testSelectedMultipleKeyspaces()
+    {
+        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool(SUBCOMMAND, "system_traces", "system_auth");
+        tool.assertOnCleanExit();
+        assertThat(tool.getStdout()).contains("Keyspace : system_traces");
+        assertThat(tool.getStdout()).contains("Keyspace : system_auth");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "Keyspace :")).isEqualTo(2);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tTable :")).isGreaterThan(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tPaths :")).isGreaterThan(1);
+    }
+
+    @Test
+    public void testSelectedTable()
+    {
+        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool(SUBCOMMAND, "system_auth.roles");
+        tool.assertOnCleanExit();
+        assertThat(tool.getStdout()).contains("Keyspace : system_auth");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "Keyspace :")).isEqualTo(1);
+        assertThat(tool.getStdout()).contains("Table : roles");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tTable :")).isEqualTo(1);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tPaths :")).isEqualTo(1);
+    }
+
+    @Test
+    public void testSelectedMultipleTables()
+    {
+        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool(SUBCOMMAND, "system_auth.roles", "system_auth.role_members");
+        tool.assertOnCleanExit();
+        assertThat(tool.getStdout()).contains("Keyspace : system_auth");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "Keyspace :")).isEqualTo(1);
+        assertThat(tool.getStdout()).contains("Table : roles");
+        assertThat(tool.getStdout()).contains("Table : role_members");
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tTable :")).isEqualTo(2);
+        assertThat(StringUtils.countMatches(tool.getStdout(), "\tPaths :")).isEqualTo(2);
+    }
+
+    @Test
+    public void testFormatArgJson()
+    {
+        ToolRunner.ToolResult tool = ToolRunner.invokeNodetool(SUBCOMMAND, "--format", "json");
```

Comment:
for both json and yaml output formats, I would get the output as string and then I would put it back to some json lib to parse it so we would eat what we produced - doing full circle here. Hence we would be sure that what we produced makes sense and it is valid output.

Reply:
This is done in [StatsPrinterTest.java](https://github.com/apache/cassandra/blob/trunk/test/unit/org/apache/cassandra/tools/nodetool/stats/StatsPrinterTest.java), isn't it?

Reply:
Probably, we can leave it out, yes. I think that is better place to do these kinds of tests.